tensorflow/lite/micro/kernels/add_common.cc

```diff
@@ -90,6 +90,11 @@ TfLiteStatus AddPrepare(TfLiteContext* context, TfLiteNode* node) {
   TfLiteTensor* output =
       micro_context->AllocateTempOutputTensor(node, kAddOutputTensor);
   TF_LITE_ENSURE(context, output != nullptr);
+  // This is to ensure that for int8/int16 output type,output does not have
+  // a "no-quantization" quantization type.
+  if (output->type == kTfLiteInt8 || output->type == kTfLiteInt16) {
+    TFLITE_CHECK_NE(output->quantization.type, kTfLiteNoQuantization);
```

Comment:
I'd suggest moving this into `CalculateOpDataAdd` in Line 42 for a few long term maintenance reasons:

 * The assert is closer to the code that assumes quantization.
 * One fewer branch to maintain for checking output types. If someone extends the branch logic in `CalculateOpDataAdd`, it will only need to be changed in on place.

Reply:
Thank you for the suggestion! Made the change.